src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableList_1.Builder.cs

```diff
@@ -750,6 +750,103 @@ public int RemoveAll(Predicate<T> match)
                 return count - this.Count;
             }
 
+            /// <summary>
+            /// Removes the specified value from this list.
+            /// </summary>
+            /// <param name="value">The value to remove.</param>
+            /// <param name="equalityComparer">
+            /// The equality comparer to use in the search.
+            /// If <c>null</c>, <see cref="EqualityComparer{T}.Default"/> is used.
+            /// </param>
+            public void Remove(T value, IEqualityComparer<T>? equalityComparer)
+            {
+                int index = this.IndexOf(value, 0, this.Count, equalityComparer);
+                if (index >= 0)
+                {
+                    this.RemoveAt(index);
+                }
+            }
+
+            /// <summary>
+            /// Removes the specified values from this list.
+            /// </summary>
+            /// <param name="index">The starting index to begin removal.</param>
+            /// <param name="count">The number of elements to remove.</param>
+            public void RemoveRange(int index, int count)
+            {
+                Requires.Range(index >= 0 && index <= this.Count, nameof(index));
+                Requires.Range(count >= 0 && index + count <= this.Count, nameof(count));
+
+                int remaining = count;
+                while (remaining-- > 0)
+                {
+                    this.RemoveAt(index);
+                }
+            }
+
+            /// <summary>
+            /// Removes the specified values from this list.
+            /// </summary>
+            /// <param name="items">The items to remove if matches are found in this list.</param>
+            /// <param name="equalityComparer">
+            /// The equality comparer to use in the search.
+            /// If <c>null</c>, <see cref="EqualityComparer{T}.Default"/> is used.
+            /// </param>
+            public void RemoveRange(IEnumerable<T> items, IEqualityComparer<T>? equalityComparer)
+            {
+                Requires.NotNull(items, nameof(items));
+
+                foreach (T item in items.GetEnumerableDisposable<T, Enumerator>())
+                {
+                    int index = this.Root.IndexOf(item, equalityComparer);
+                    if (index >= 0)
+                    {
+                        this.RemoveAt(index);
+                    }
+                }
+            }
+
+            /// <summary>
+            /// Removes the specified values from this list.
+            /// </summary>
+            /// <param name="items">The items to remove if matches are found in this list.</param>
+            public void RemoveRange(IEnumerable<T> items)
+            {
+                this.RemoveRange(items, EqualityComparer<T>.Default);
+            }
```

Comment:
Use expression style if possible (everywhere in this code)
```suggestion
            public void RemoveRange(IEnumerable<T> items) => RemoveRange(items, EqualityComparer<T>.Default);
```